Mathlib/Lean/Expr.lean

```diff
@@ -0,0 +1,117 @@
+/-
+Copyright (c) 2019 Robert Y. Lewis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro, Simon Hudon, Scott Morrison, Keeley Hoek, Robert Y. Lewis, Floris van Doorn
+-/
+import Lean
+/-!
+# Additional operations on Expr and related types
+
+This file defines basic operations on the types expr, name, declaration, level, environment.
+
+This file is mostly for non-tactics.
+-/
+
+open Lean Meta
+
+namespace Lean
+
+namespace BinderInfo
+
+/-! ### Declarations about `BinderInfo` -/
+
+
+instance : Inhabited BinderInfo :=
+  ⟨BinderInfo.default⟩
+
+/-- The brackets corresponding to a given `BinderInfo`. -/
+def brackets : BinderInfo → String × String
+| BinderInfo.implicit => ("{", "}")
+| BinderInfo.strictImplicit => ("{{", "}}")
+| BinderInfo.instImplicit => ("[", "]")
+| _ => ("(", ")")
+
+end BinderInfo
+
+/-- The type of binders containing a name, the binder info and the binding type -/
+structure Binder :=
+  (name : Name)
+  (info : BinderInfo)
+  (type : Expr)
+deriving Inhabited
```

Comment:
```suggestion
structure Binder where
  name : Name
  info : BinderInfo
  type : Expr
  deriving Inhabited
```

I think this structure should be deleted completely.  The idiomatic way is to use free variables (mkFVar) and then look up the user name, binder info, type, etc. in the local context.

The Binder structure doesn't do let bindings either.